### PR TITLE
Update ni cluster config for connection

### PR DESCRIPTION
**Pull request by hunner**
For when the k8s connection lands.

### CODE REVIEW

File: pkg/task/cluster.go [hunk 1]
```diff
@@ -27,11 +28,31 @@ func (ti *TaskInterface) ProcessClusters(directory string) error {
 			return err
 		}
 	}
+	if cluster == nil {
```
Comment (impl) [inline pkg/task/cluster.go:31]: It doesn't look like `retrieveClusterFromEnvironment` can ever return a nil value, can this check go away?

File: pkg/task/cluster.go [hunk 2]
```diff
@@ -41,12 +62,12 @@ func CreateKubeconfigFile(directory string, resource *model.ClusterDetails) erro
...
 	cluster := &clientcmdapi.Cluster{
-		Server:                resource.URL,
+		Server:                resource.Connection.Server,
 		InsecureSkipTLSVerify: resource.Insecure,
 	}
 
-	if resource.CAData != "" {
-		ca, err := base64.StdEncoding.DecodeString(resource.CAData)
+	if resource.Connection.CertificateAuthority != "" {
+		ca, err := base64.StdEncoding.DecodeString(resource.Connection.CertificateAuthority)
```
Comment (impl) [inline pkg/task/cluster.go:70]: This decoding needs to happen at the conversion time to the connection above. User input into the connection form won't be base64-encoded.

File: pkg/task/cluster.go [hunk 3]
```diff
@@ -93,13 +114,13 @@ func retrieveClusterFromEnvironment() *model.ClusterDetails {
 		cluster.Name = nameFromEnv
 	}
 	if urlFromEnv := os.Getenv("CLUSTER_URL"); urlFromEnv != "" {
-		cluster.URL = urlFromEnv
+		cluster.Connection.Server = urlFromEnv
```
Comment (impl) [inline pkg/task/cluster.go:117]: `cluster.Connection` will be nil here I think. I'm not actually sure if this code path is used anywhere, though. We might be able to remove this method entirely?

Comment (hunner) [reply pkg/task/cluster.go:117]: Yeah sure, if this isn't used then I'd be happy to remove it. I don't see any occurrences of these env vars in our github org.

File: pkg/task/cluster.go (hunk 2 above)
Comment (hunner) [reply pkg/task/cluster.go:70]: Ah yes! Thanks for this.

File: pkg/task/cluster.go [hunk 4]
```diff
@@ -20,9 +18,9 @@ func (ti *TaskInterface) ProcessClusters(directory string) error {
 		return err
 	}
 
-	cluster := retrieveClusterFromEnvironment()
+	cluster := &model.ClusterDetails{}
 
-	if cluster != nil && spec.Cluster != nil {
+	if spec.Cluster != nil {
```
Comment (impl) [inline pkg/task/cluster.go:23]: You could probably simplify this to not use `mergo`, which is pretty heavyweight. I.e.:

```
cluster := spec.Cluster
if cluster == nil {
  cluster = &model.ClusterDetails{}
}
```

File: pkg/task/cluster.go [hunk 5]
```diff
@@ -32,6 +30,27 @@ func (ti *TaskInterface) ProcessClusters(directory string) error {
...
+		cluster.Connection = &model.ClusterConnectionSpec{}
+	}
+	if cluster.Connection.Server == "" {
+		cluster.Connection.Server = cluster.URL
+		cluster.URL = ""
+	}
+	if cluster.Connection.CertificateAuthority == "" {
+		ca, err := base64.StdEncoding.DecodeString(cluster.CAData)
+		if err != nil {
+			return err
```
Comment (impl) [inline pkg/task/cluster.go:44]: I believe you might get an error if `cluster.CAData` is empty in this case. Might want to make sure that lines up with the previous behavior.

Comment (hunner) [reply pkg/task/cluster.go:44]: Looks fine. Decoding empty `string` is just an empty `[]byte` and results in no `certificate-authority-data` in the kubectl config.

**Pull request merged**